Функционал основного сервиса

ScoobyWraith opened the pull request.

Verba163 committed:
Сделать скелет будущего проекта

Verba163 committed:
Сделать скелет будущего проекта

Verba163 committed:
Сделать скелет будущего проекта

Verba163 committed:
Сделать скелет будущего проекта

Verba163 committed:
Сделать скелет будущего проекта

Verba163 committed:
Сделать скелет будущего проекта

MaxSWR committed:
[Sprint 17][main-service] Add config for h2

MaxSWR committed:
[Sprint 17][main-service] Add dependencies: h2, QueryDSL

MaxSWR committed:
[Sprint 17][main-service] Remove events mapping (temp)

MaxSWR committed:
[Sprint 17][main-svc-events] Implement events controller

ScoobyWraith committed:
Merge pull request #11 from Verba163/main_svc_events

Implement events controller

Verba163 committed:
Настроить Docker и БД Создать юзера

Verba163 committed:
Настроить Docker и БД Создать юзера

Verba163 committed:
Merge remote-tracking branch 'origin/main_svc_user' into main_svc_user

# Conflicts:
#	main-service/pom.xml
#	main-service/src/main/java/ru/practicum/ewm/MainApplication.java
#	main-service/src/main/java/ru/practicum/ewm/compilation/mapper/CompilationMapper.java
#	main-service/src/main/java/ru/practicum/ewm/request/dto/EventRequestStatusUpdateResultDto.java

Verba163 committed:
Merge pull request #12 from Verba163/main_svc_user

Настроить Docker и БД Создать юзера

MaxSWR committed:
Merge remote-tracking branch 'origin/main_svc' into main_svc_events

MaxSWR committed:
[Sprint 17][main-svc-events] Update event model

Tadaraki-code committed:
Add category

Tadaraki-code committed:
Merge pull request #13 from Verba163/main_svc_category

Add category

MaxSWR committed:
[Sprint 17][main-svc-events] Update EventsServiceImpl [part 1]

MaxSWR committed:
Merge remote-tracking branch 'origin/main_svc' into main_svc_events

MaxSWR committed:
[Sprint 17][main-svc-events] Update EventsServiceImpl [part 2]

Verba163 committed:
Реквесты и эксепшен

Verba163 committed:
Merge remote-tracking branch 'origin/main_svc' into main_svc_request

# Conflicts:
#	main-service/src/main/java/ru/practicum/ewm/error/exception/ConflictException.java
#	main-service/src/main/java/ru/practicum/ewm/exception/GlobalExceptionHandler.java
#	main-service/src/main/resources/application.yml

Verba163 committed:
Merge pull request #15 from Verba163/main_svc_request

Main svc request

MaxSWR committed:
Merge remote-tracking branch 'origin/main_svc' into main_svc_events

# Conflicts:
#	main-service/src/main/java/ru/practicum/ewm/events/storage/EventsRepository.java

MaxSWR committed:
[Sprint 17][main-svc-events] Remove old exception

MaxSWR committed:
[Sprint 17][main-svc-events] Add countByCategoryId + move controller

ScoobyWraith committed:
Merge pull request #14 from Verba163/main_svc_events

Обновление для ивентов

MaxSWR committed:
Merge remote-tracking branch 'origin/main_svc' into main_svc_events

Verba163 committed:
Изменен Юзер и Эксепшены

Verba163 committed:
Merge pull request #17 from Verba163/main_svc_exception

Изменен Юзер и Эксепшены

MaxSWR committed:
[Sprint 17][main-svc-events] Update EventsServiceImpl [part 3]

MaxSWR committed:
Merge remote-tracking branch 'origin/main_svc' into main_svc_events

# Conflicts:
#	main-service/src/main/java/ru/practicum/ewm/events/service/EventsServiceImpl.java

MaxSWR committed:
[Sprint 17][main-svc-events] Postman tests: complete validation events

Verba163 committed:
Refactor user query parameters to use a dedicated class

Introduced `UserQueryParams` to encapsulate user query parameters, improving code clarity and reusability. Replaced individual parameters in `getAllUsers` methods across the service, controller, and repository layers with this new class. This change helps streamline parameter validation and handling.

MaxSWR committed:
[Sprint 17][main-svc-events] Postman tests: complete events

MaxSWR committed:
Merge remote-tracking branch 'origin/main_svc' into main_svc_events

# Conflicts:
#	main-service/src/main/java/ru/practicum/ewm/user/controller/UserController.java

Verba163 committed:
Refactor user query parameters to use a dedicated class

Introduced `UserQueryParams` to encapsulate user query parameters, improving code clarity and reusability. Replaced individual parameters in `getAllUsers` methods across the service, controller, and repository layers with this new class. This change helps streamline parameter validation and handling.

ScoobyWraith committed:
Merge pull request #18 from Verba163/main_svc_events

Ивенты: состояние, с которым проходятся postman тесты Event и Validation.Event

Verba163 committed:
Add request validation logic and improve exception handling

Introduced `RequestValidator` to centralize request validation logic, ensuring consistent rules such as participant limits and event states. Updated repositories and services to utilize the new validator. Standardized exception handling with refined `NotFoundException` and corrected package references for `ConflictException`.

Verba163 committed:
Merge pull request #19 from Verba163/main_svc_requests

Add request validation logic and improve exception handling

Verba163 committed:
Refactor to improve constants usage and validation logic

Centralized constants for requests and users to dedicated classes to improve maintainability. Enhanced request validation logic by accurately counting confirmed participants. Replaced ValidationException with a more consistent DataIntegrityViolationException in event service for better error handling.

Tadaraki-code committed:
Add Compilation

Tadaraki-code committed:
Add Compilation

Tadaraki-code committed:
Merge pull request #20 from Verba163/main_svc_compilation

Add Compilation

Verba163 committed:
Refactor to improve constants usage and validation logic

Refactor to improve constants usage and validation logic

Centralized constants for requests and users to dedicated classes to improve maintainability. Enhanced request validation logic by accurately counting confirmed participants. Replaced ValidationException with a more consistent DataIntegrityViolationException in event service for better error handling.

Tadaraki-code committed:
Add handleConflictException

MaxSWR committed:
Merge remote-tracking branch 'origin/main_svc' into main_svc_events

Verba163 committed:
Update stat-service host and refine docker-compose config

Changed stat-service host to 'stats-server' in application.yml for environment alignment. Updated docker-compose to include stats-server dependency and adjusted CLIENT_URL path for proper routing.

Verba163 committed:
Merge pull request #23 from Verba163/main_svc_hit_patch

Update stat-service host and refine docker-compose config

Tadaraki-code committed:
Merge pull request #22 from Verba163/main_svc_category_patch

Add handleConflictException

MaxSWR committed:
Merge remote-tracking branch 'origin/main_svc' into main_svc_events

Tadaraki-code committed:
Patch docker

Tadaraki-code committed:
Merge pull request #24 from Verba163/main_svc_docker_patch

Patch docker

MaxSWR committed:
[Sprint 17][main-svc-events] Postman tests: complete all tests [part 1# хе хе]

MaxSWR committed:
Merge remote-tracking branch 'origin/main_svc' into main_svc_events

ScoobyWraith committed:
Merge pull request #25 from Verba163/main_svc_events

Main svc events

MaxSWR committed:
Merge remote-tracking branch 'origin/main' into main_svc

# Conflicts:
#	docker-compose.yml

MaxSWR committed:
Merge remote-tracking branch 'origin/main_svc' into main_svc_events

MaxSWR committed:
[Sprint 17][main-svc-events] en language + move classes/enums into packages

ScoobyWraith committed:
Merge pull request #27 from Verba163/main_svc_events

Main svc events

felixoff commented on main-service/src/main/java/ru/practicum/ewm/category/dto/CategoryParams.java:
```diff
@@ -0,0 +1,17 @@
+    private Long catId;
+
+    private Long from;
+
```
Используй @FieldDefaults вместо множественных private

felixoff commented on main-service/src/main/java/ru/practicum/ewm/category/service/CategoryServiceImpl.java:
```diff
@@ -0,0 +1,137 @@
+        checkAndGetCategory(catId);
+
+        if (eventsRepository.countByCategoryId(catId) > 0) {
+            throw new ConflictException("Category with id '" + catId + "' is not empty");
```
Используй String.format

felixoff commented on main-service/src/main/java/ru/practicum/ewm/events/controller/EventsController.java:
```diff
@@ -0,0 +1,225 @@
+        return eventsService.updateEvent(updateEventParameters);
+    }
+
+    @GetMapping(EventsConstants.PRIVATE_API_PREFIX + "/{userId}/events/{eventId}/requests")
```
Пути принято писать через дефис

Verba163 committed:
Simplify imports and enhance code consistency.

Replaced explicit imports with wildcard imports in controllers and DTOs for conciseness. Refactored repeated path constants into reusable variables. Standardized exception messages using `String.format` and applied consistent field visibility with Lombok's `@FieldDefaults`.

Verba163 committed:
Fix formatting issues in constants and controller files

Corrected minor spacing issues in `EventsConstants` and `CategoryController` classes to improve code readability and maintain consistent formatting. No functional changes were made.
